pkg/devserver/devserver.go

```diff
@@ -857,24 +793,36 @@ func NormalizeThrottle(smv2 sv2.StateLoader, dbcqrs cqrs.Manager) redis_state.Re
 }
 
 func PartitionConstraintConfigGetter(dbcqrs cqrs.Manager) redis_state.PartitionConstraintConfigGetter {
-	return func(ctx context.Context, p redis_state.QueueShadowPartition) (*redis_state.PartitionConstraintConfig, error) {
-		if p.EnvID == nil || p.FunctionID == nil {
-			return &redis_state.PartitionConstraintConfig{
-				Concurrency: redis_state.ShadowPartitionConcurrency{
-					SystemConcurrency:   consts.DefaultConcurrencyLimit,
-					AccountConcurrency:  consts.DefaultConcurrencyLimit,
-					FunctionConcurrency: consts.DefaultConcurrencyLimit,
+	return func(ctx context.Context, p redis_state.PartitionIdentifier) redis_state.PartitionConstraintConfig {
+		if p.SystemQueueName != nil {
+			return redis_state.PartitionConstraintConfig{
+				Concurrency: redis_state.PartitionConcurrency{
+					SystemConcurrency:   consts.DefaultSystemConcurrencyLimit,
+					AccountConcurrency:  consts.DefaultSystemConcurrencyLimit,
+					FunctionConcurrency: consts.DefaultSystemConcurrencyLimit,
 				},
-			}, nil
+			}
 		}
 
-		workflow, err := dbcqrs.GetFunctionByInternalUUID(ctx, *p.EnvID, *p.FunctionID)
+		workflow, err := dbcqrs.GetFunctionByInternalUUID(ctx, p.EnvID, p.FunctionID)
```

Comment:
not that it matter for this PR, but this is a weird interface.
the ID of the function should be indexed already so unclear why it needs the envID at all.

Reply:
it doesn’t actually, i temporarily removed it but thought well let’s keep it if we want multiple envs for self-hosting or something. but even then the fnID is globally unique

Reply:
yeah exactly that's kind of the point of having a UUID

Reply:
i mean sometimes we have compound keys but not here 😀